File: Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        string temp;
        int l = 0 , r = 0,i = 0;
        int maxs=0;
        while( r < s.size() && l < s.size())
        {
            if(find(temp.begin(), temp.end(),s[r]) == temp.end())
                {
                    temp.push_back(s[r]);
                    r++;
                }
            else 
                {
                    l++;
                    r = l;
                    temp.clear();
                }
                 maxs = max(maxs,r - l);
        }
        return maxs;
    }
};

File: longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        a = set()
        for i in s:
            if i in a:
                return len(a)
            a.add(i)
        return len(a)


if __name__ == '__main__':
    print(Solution().lengthOfLongestSubstring("pwwkew"))
